src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -52,10 +52,95 @@ private void WriteClientMethodRequest(CodeWriter writer, LowLevelClientMethod cl
         }
 
         private void WriteClientMethod(CodeWriter writer, LowLevelClientMethod clientMethod, bool async)
+        {
+            WriteClientMethodDecleration(writer, clientMethod, async);
+
+            if (clientMethod.Operation.IsLongRunning)
+            {
+                var finalStateVia = clientMethod.Operation.LongRunningFinalStateVia;
+
+                WriteClientMethodBody(writer, clientMethod, async, (writer) =>
+                {
+                    writer.Line($"return new LowLevelBinaryDataOperation(_clientDiagnostics, Pipeline, message.Request, message.Response, {typeof(OperationFinalStateVia)}.{finalStateVia}, {clientMethod.Diagnostics.ScopeName:L});");
+                });
+            }
+            else
+            {
+                WriteClientMethodBody(writer, clientMethod, async, (writer) =>
+                {
+                    writer.Line($"return message.Response;");
+                });
+            }
+        }
+
+        private void WriteClientMethodBody(CodeWriter writer, LowLevelClientMethod clientMethod, bool async, Action<CodeWriter> returnWriter)
```

Comment:
Can we make `returnWriter` a formattable string rather than a delegate?

Reply:
We sure can.  Updated.